client/modules/IDE/pages/IDEView.jsx

```diff
@@ -173,6 +173,14 @@ class IDEView extends React.Component {
       // don't warn
       this.props.persistState();
       window.onbeforeunload = null;
+    } else if (route && (/[\s\S]*\/sketches\/?$/.test(route.pathname))) {
+      // don't warn
+      this.props.persistState();
+      window.onbeforeunload = null;
+    } else if (route && (route.pathname === this.props.ide.previousPath)) {
```

Comment:
what's this if statement checking?

Reply:
Now suppose you click on my sketches and you have ```unsavedChanges``` to be true so when you click to open another sketch name it will display a warning but if you click the cross button to close the modal no warning should be displayed. if condition is checking if you are going to the same page which you came from.

Reply:
got it! that makes sense.